src/eos-updater-fetch.c

```diff
@@ -113,27 +139,30 @@ content_fetch (GTask *task,
       goto error;
     }
 
-  message ("Fetch: %s:%s resolved to: %s", src, ref, sum);
+  message ("Fetch: %s:%s resolved to: %s", remote, ref, sum);
+  progress = ostree_async_progress_new_and_connect (update_progress, updater);
+
+  if (data->overridden_urls)
+    {
+      guint idx = g_random_int_range (0, g_strv_length (data->overridden_urls));
 
+      url_override = data->overridden_urls[idx];
```

Comment:
This is what was proposed for now. In the future there was an idea of fetching objects from several URLs in parallel, but that would require changes in the OSTree pulling code.

What could be done here later is to try all the URLs one by one when fetching fails for some reason.